rdr_service/dao/base_dao.py

```diff
@@ -774,7 +774,7 @@ def _validate_update(self, session, obj, existing_obj):
     that it matches.
     """
         if not existing_obj:
-            raise NotFound(f"{self.model_type.__name__} with id {id} does not exist")
+            raise NotFound(f"{self.model_type.__name__} with id {getattr(obj, 'id', id)} does not exist")
```

Comment:
I initially updated this line to use `obj.id`. But Participant DAO gave an error because the participant table does not have a column named `id`. I’ve updated the code to check for the presence of an id in the model. If it exists, the code will use that; if not, it will continue with the previous behavior.